### PR TITLE
sfml: new recipe

### PR DESCRIPTION
**Pull request by kenmays**
SFML 2.5.1 for Haiku.
NOTE: Builds on Haiku R1B2 - Closes https://github.com/haikuports/haikuports/issues/4469
Part of: https://github.com/CodeforEvolution/SFML/tree/feature/haiku_port

### CODE REVIEW

**Comment by CodeforEvolution** [comment]
Thank you for writing this recipe! :D As you've mentioned, most of the library builds just fine at this point and is functional.

However, what's left to port (and is the biggest portion to port of course) is the Window module. Despite being called the Window module, the module not only contains window management functionality, but also the controlling and monitoring of input devices, clipboard handling, handling of the cursor, and the handling of video modes. I'm actually quite excited for the cursor portion, as I believe custom cursors were just implemented in Haiku! :)

**Comment by kenmays** [comment]
Builds and packages. I need to fine tune the packaging.

**Comment by kenmays** [comment]
Ready for Merge .

File: media-libs/sfml/sfml-2.5.1.recipe [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	sfml$secondaryArchSuffix = $portVersion compat >= 1
+	lib:libsfml$secondaryArchSuffix = 2.5.1 compat >= 0
+	lib:libsfml_audio$secondaryArchSuffix = 2.5.1 compat >= 0
+	lib:libsfml_graphics$secondaryArchSuffix = 2.5.1 compat >= 0
+	lib:libsfml_network$secondaryArchSuffix = 2.5.1 compat >= 0
+	lib:libsfml_system$secondaryArchSuffix = 2.5.1 compat >= 0
+	lib:libsfml_window$secondaryArchSuffix = 2.5.1 compat >= 0
+	"
+
+REQUIRES="
```
Comment (diversys) [inline media-libs/sfml/sfml-2.5.1.recipe:30]: Sort alphabetically here and in other sections.

**Comment by diversys** [comment]
Drop patches/readme.md

**Comment by extrowerk** [comment]
As long as the port incomplete this shouldn't be merged.

**Comment by Begasus** [comment]
@CodeforEvolution could you shed a light on this? Is this in any way ready to be used?

**Comment by kenmays** [comment]
@CodeforEvolution - I'm resubmitting an updated PR, so this PR preserves the patchset and recipe for now. I'll rebase updates for the Window subsystem.

**Pull request closed without merging**